cms/djangoapps/api/docs/decisions/001-course-apps-api.rst

```diff
@@ -0,0 +1,153 @@
+Course Apps API
+_______________
+
+Status
+======
+Proposal
+
+Context
+=======
+
+The new `Course Authoring MFE`_ includes a new UX called "Pages and Resources"
+for configuring course "apps" such as progress, wiki, teams, discussions etc.
+all from one place. Currently these apps are intermingled with the core
+platform and are inseparable from it, however future apps could be implemented
+as plugins.
+
+In order to drive this new UI, we need a way to discover installed and available
+course apps and provide an API that can be used by MFEs to display information
+related to these apps and potentially enable/disable these apps.
+
+.. _Course Authoring MFE: https://github.com/edx/frontend-app-course-authoring/
+
+
+Decision
+========
+
+We can develop a new course apps API that can be used to manage the apps
+available for a course. The list of *available* apps might not match the list
+of *installed* apps since it is possible that certain apps may be disabled, or
+otherwise unavailable for a particular course.
+
+The course apps API will only list the available apps, and allow toggling them
+on or off. It will not support any other verbs or settings. Each app can
+provide its own API(s) for configuring app-specific settings, which will not be
+in scope here. Currently the mechanism for extending the frontend with custom
+config pages for plug-in apps is TBD.
+
+Each app will have the following associated metadata:
+
+- **id**: A unique identifier for the app. This can be used in the frontend to
+  look up the app's friendly title and description.
+- **enabled**: Is this app enabled for the current course.
+- **permissions**: Apps can potentially enable/disable certain operations, for
+  users based on their role. The permissions a user has can be listed here. Apps
+  may have their own permissions but at this very least this will include:
+
+    - **enable**: Can the current user enable/disable this app.
+    - **configure**: Can the current user can configure this app.
+
```

Comment:
I'm not sure if the permissions mechanism should be more formalised. Perhaps we can ask each plugin to expose the permissions as a bridegekeeper rules? This way we ensure that core platform has a better knowledge of the available permissions.

Reply:
What's the use case for needing the permissions?  Not sure I understand the circumstances in which a user in this UI would _not_ have permission to enable/disable or configure an app. 

That said, I think this sub-object for permissions is a fine structure for it, assuming we need it.

Reply:
The reason for adding this is that I can imagine how disabling an app mid-course can be a destructive operation that could cause major issues, so that the operation can be disabled for people who would otherwise be able to enable the app in the first place. 

I think perhaps changing the name to "allowed_operation" might be more generic? i.e. permissions aren't the only thing stopping someone from performing an operation. 

For instance, there may be core apps that can't be disabled. Or apps that don't have any configuration at all. In that case we need a way to signal to the UI to not show the related operations.